Buildings/Resources/src/ThermalZones/Detailed/Validation/BESTEST/simulateAndPlot.py

```diff
@@ -0,0 +1,988 @@
+    """ Save the figure to a pdf and png file in the directory `img`
+    """
+
+    out_dir = "../../../../../Images/ThermalZones/Detailed/Validation/BESTEST"
+    if not os.path.exists(out_dir):
+        os.makedirs(out_dir)
+    figure.savefig(os.path.join(out_dir, '{}.pdf'.format(file_name)))
+    figure.savefig(os.path.join(out_dir, '{}.png'.format(file_name)))
+    plt.clf()
+
+def configure_axes(axes):
+    """ Configure the axis style
+    """
+    axes.spines['right'].set_linewidth(0.25)
+    axes.spines['right'].set_color('lightgrey')
+    axes.spines['top'].set_visible(False)
+    axes.spines['left'].set_linewidth(0.25)
+    axes.spines['left'].set_color('lightgrey')
+    axes.spines['bottom'].set_visible(False)
+    axes.grid(color='lightgrey', linewidth=0.25)
+    return
+
+def create_library_directory(repoNam):
+    ''' Create directory to be as temporary directory to save buildings library.
+    '''
+    import tempfile
+    import getpass
+    prefixName = 'tmp-MBL-' if repoNam == 'MBL' else 'tmp-BP-'
+    worDir = tempfile.mkdtemp( prefix=prefixName + getpass.getuser() )
+    print("Created directory {}".format(worDir))
+    return worDir
+
+def checkout_repository(working_directory, repoNam, from_git_hub, branch):
+    ''' Clone or copy repository to working directory.
+    '''
+    from git import Repo
+    import git
+    d = {}
+    if from_git_hub:
+        print("Checking out repository branch %s" % branch)
+        git_url = "https://github.com/lbl-srg/modelica-buildings" if repoNam == 'MBL' else "https://github.com/lbl-srg/BuildingsPy.git"
+        r = Repo.clone_from(git_url, working_directory)
+        g = git.Git(working_directory)
+        g.checkout(branch)
+        # Print commit
+        d['branch'] = branch
+        d['commit'] = str(r.active_branch.commit)
+    else:
+        mbl_local_path = "../../../../../../../../modelica-buildings"
+        bp_local_path = "../../../../../../../../BuildingsPy"
+        # This is a hack to get the local copy of the repository
+        des = working_directory
+        shutil.rmtree(des)
+        print("*** Copying library to {}".format(des))
+        localPath = mbl_local_path if repoNam == 'MBL' else bp_local_path
+        shutil.copytree(localPath, des)
+        g = git.Git(des)
+        g.checkout(branch)
+    return d
+
+def _runTests(tool, package, lib_dir, n_pro, bp_dir):
+    ''' Run regression test, return the temporary directory path for the test.
+    '''
+    import sys
+    regTestPath = os.path.join(bp_dir, 'buildingspy', 'development')
+    sys.path.append(regTestPath)
+    import regressiontest as u
+
+    ut = u.Tester(tool=tool, skip_verification=True)
+    ut.batchMode(True)
+    ut.pedanticModelica(True)
+    ut.showGUI(False)
+    ut.deleteTemporaryDirectories(False)
+
+    path = os.path.join(lib_dir, 'Buildings')
+
+    ut.setLibraryRoot(path)
+    if package is not None:
+        ut.setSinglePackage(package)
+    ut.setNumberOfThreads(n_pro)
```

Comment:
no need to set `n_pro` if you want to use all available processors.

Reply:
Updated. `n_pro` was removed.